jina/drivers/rank/__init__.py

```diff
@@ -49,11 +48,10 @@ def _apply_all(self, docs: 'DocumentSet', context_doc: 'Document', *args,
             - Set the ``traversal_paths`` of this driver such that it traverses along the ``matches`` of the ``chunks`` at the level desired.
         """
 
-        # if at the top-level already, no need to aggregate further
-        query_meta = context_doc.get_attrs(*self.exec.required_keys)
-
+        query_meta = context_doc.get_attrs(*self.exec.required_keys) if hasattr(self.exec, 'required_keys') else None
```

Comment:
Why this `if/else`? Why is it needed out of a sudden?

Reply:
My bad, this is an advancement of an idea I had, but I will wait for another PR.

I was thinking about allowing to have different set of `required_keys` for `query` and for `matches`, and therefore some of the rankers may not need one of the sets. 

I will remove for now

Reply:
I also think is not very good to have `text` as default. I'd rather have the class as abstract than having such a strong default

Reply:
But good point, removed now